gtsfm/utils/ellipsoid.py

```diff
@@ -0,0 +1,103 @@
+"""Algorithms to fit Ellisoid geometry to 3D points and extract the rotation matrix necessary to align point cloud 
+with x, y, and z axes. Used in React Three Fiber Visualization Tool.
+
+Authors: Adi Singh
+"""
+
+import numpy as np
+from numpy.linalg import inv, norm, svd
+
+
+def center_point_cloud(point_cloud: np.ndarray) -> np.ndarray:
+    """Centers a point cloud using mean values of x, y, and z.
+
+    Args:
+        pointCloud: array representing the point cloud, of shape N x 3.
+
+    Returns:
+        points_centered: centered point cloud of shape Nx3.
+        means: the means of the x,y,z coordinates, array of length 3.
+
+    Raises:
+        TypeError: if points list is not 3 dimensional.
+    """
+    if point_cloud.shape[1] != 3:
+        raise TypeError("Points list should be 3D")
+
+    means = np.mean(point_cloud, axis=0)
+    points_centered = point_cloud - means
+    return points_centered, means
+
+
+def filter_outlier_points(centered_pc: np.ndarray) -> np.ndarray:
+    """Filters the top 5% of points with greatest distance from origin.
+
+    Args:
+        centered_pc: centered point cloud of shape N x 3.
+
+    Returns:
+        The filtered point cloud of shape M x 3 (M = 0.95*N).
+
+    Raises:
+        TypeError: if centered point cloud is not 3 dimensional.
+    """
+    if centered_pc.shape[1] != 3:
+        raise TypeError("Point Cloud should be 3 dimensional")
+
+    # find mean and standard deviation of magnitudes of points to determine the threshold magnitude to filter by
+    mags = norm(centered_pc, axis=1)
+    mean_mag = np.mean(mags)
+    std_mag = np.std(mags)
+
+    # z = 1.65 represents the 95th percentile. Assuming a normal distribution of point magnitudes.
+    cutoff_mag = 1.65 * (std_mag) + mean_mag
```

Comment:
one other approach would be `np.percentile(ranges, 95)`. Which is better? i'm not sure

Reply:
oh.. I didn't even think to check if np had a percentile function. I'll look into it a bit closer to see if it's cleaner.

Reply:
actually, i think doing `np.percentile(ranges, 95)` would not be simpler since it only takes the percentile of individual numbers in the array. Whereas in our case, we wish to take the 95th percentile of the magnitudes of each 3d point. So I'll probably leave this `cutoff_mag` code as it is.

Reply:
Would it be identical to 3 lines
```python
ranges = np.linalg.norm(point_cloud, axis=1)
cutoff_range = np.percentile(ranges, 95)
point_cloud = point_cloud[ mags < cutoff_range]
```